src/main/java/com/google/devtools/build/lib/rules/cpp/CcModule.java

```diff
@@ -396,9 +400,22 @@ public LibraryToLink createLibraryLinkerInput(
     Artifact dynamicLibrary = nullIfNone(dynamicLibraryObject, Artifact.class);
     Artifact interfaceLibrary = nullIfNone(interfaceLibraryObject, Artifact.class);
 
-    Artifact notNullArtifactForIdentifier = null;
     StringBuilder extensionErrorsBuilder = new StringBuilder();
     String extensionErrorMessage = "does not have any of the allowed extensions";
```

Comment:
dead code

Reply:
It's still used in the validation logic for the static and dynamic library attributes. E.g. [here](https://github.com/bazelbuild/bazel/pull/8888/files/43dc9a749595ab764c65eb7dd42122e2b5460899#diff-f321a84e135b6943c822c7a94df47b82R428). Sadly, it doesn't match so well with the symlink validation and refactoring that code too much seems out of scope for this PR.